alcs-api/src/application/application.controller.ts

```diff
@@ -132,27 +132,54 @@ export class ApplicationController {
       assigneeUuid: application.assigneeUuid,
       paused: application.paused,
       highPriority: application.highPriority,
+      datePaid: this.formatIncomingDate(application.datePaid),
+      dateAcknowledgedIncomplete: this.formatIncomingDate(
+        application.dateAcknowledgedIncomplete,
+      ),
+      dateReceivedAllItems: this.formatIncomingDate(
+        application.dateReceivedAllItems,
+      ),
+      dateAcknowledgedComplete: this.formatIncomingDate(
+        application.dateAcknowledgedComplete,
+      ),
     });
 
     if (
-      app.assigneeUuid !== existingApplication.assigneeUuid &&
-      app.assigneeUuid !== req.user.entity.uuid
+      updatedApplication.assigneeUuid !== existingApplication.assigneeUuid &&
+      updatedApplication.assigneeUuid !== req.user.entity.uuid
     ) {
       this.notificationService.createForApplication(
         req.user.entity,
-        app.assigneeUuid,
+        updatedApplication.assigneeUuid,
         "You've been assigned",
-        app,
+        updatedApplication,
       );
     }
 
-    const mappedApps = await this.applicationService.mapToDtos([app]);
-    return mappedApps[0];
+    const mappedApps = await this.applicationService.mapToDtos([
+      updatedApplication,
+    ]);
+    return {
+      ...mappedApps[0],
+      statusDetails: updatedApplication.status,
+      typeDetails: updatedApplication.type,
+      regionDetails: updatedApplication.region,
+    };
   }
 
   @Delete()
   @UserRoles(...ANY_AUTH_ROLE)
   async softDelete(@Body() applicationNumber: string): Promise<void> {
     await this.applicationService.delete(applicationNumber);
   }
+
+  private formatIncomingDate(date?: number) {
+    if (date) {
+      return new Date(date);
+    } else if (date === null) {
```

Comment:
why do you need to distinguish between null and undefined?

Reply:
I need to set null to clear the date in case of someone removing it